lesson1/src/stacks_queues/stackProblems.js

```diff
@@ -2,30 +2,38 @@ let Stack = require("./stack");
 
 class StackProblems {
 
-  /*
-   A string with balanced parenthesis is one such as all closing
-   parenthesis characters ')' match their respecting opening ones '('.
-   Note: brackets and curly braces are IGNORED
+    /*
+     A string with balanced parenthesis is one such as all closing
+     parenthesis characters ')' match their respecting opening ones '('.
+     Note: brackets and curly braces are IGNORED
 
-   This method should return true if a given String (string)
-   has balanced parenthesis.
+     This method should return true if a given String (string)
+     has balanced parenthesis.
 
-   Eg1: "hello(world)" -> true
-   Eg2: "hello(wo(rld))" -> true
-   Eg3: "hello(wo(rld))(())()" -> true
-   Eg4: "hello(wo))(" -> false
-   Eg5: "hellow)orld" -> false
-   Eg6: "hellow)orld(" -> false
-   */
-  static balancedParenthesis(string) {
-    let stack = new Stack();
-    for (let i = 0; i < string.length; i++) {
-      let char = string.charAt(i);
-      // todo: do you'r magic ! 🙌
+     Eg1: "hello(world)" -> true
+     Eg2: "hello(wo(rld))" -> true
+     Eg3: "hello(wo(rld))(())()" -> true
+     Eg4: "hello(wo))(" -> false
+     Eg5: "hellow)orld" -> false
+     Eg6: "hellow)orld(" -> false
+     */
+    static balancedParenthesis(string) {
+        let stack = new Stack();
+        for (let i = 0; i < string.length; i++) {
+            let char = string.charAt(i);
+            if (char === '(') {
+                stack.push(char);
+            }
+            if (char === ')') {
+                if (stack.length() > 0) {
+                    stack.pop()
+                } else {
+                    stack.push(char)
```

Comment:
Nice ! i think as an optimization @ this point you can just return false.. if you have a closing ')' but no previous opening one that is enough proof that its not correctly balanced